### PR TITLE
Move public asset files to the proper directory

### PR DESCRIPTION
**Pull request by wxiaoguang**
Move `public/*` to `public/assets/*`

Some old PRs (like #15219) introduced inconsistent directory system.

For example: why the local directory "public" is accessed by `http://site/assets`? How to serve the ".well-known" files properly in the public directory?

For convention rules, the "public" directory is widely used for the website's root directory. It shouldn't be an exception for Gitea.

So, this PR makes the things consistent:

* `http://site/assets/foo` means `{CustomPath}/public/assets/foo`.
* `{CustomPath}/public/.well-known` and `{CustomPath}/public/robots.txt` can be used in the future.

This PR is also a prerequisite for a clear solution for:
* #21942
* #25892 
* discourse.gitea.io: [.well-known path serving custom files behind proxy?](https://discourse.gitea.io/t/well-known-path-serving-custom-files-behind-proxy/5445/1)

This PR is breaking for users who have custom "public" files (CSS/JS). After getting approvals, I will update the documents.

----

## ⚠️ BREAKING ⚠️

If you have files in your "custom/public/" folder, please move them to  "custom/public/assets/".

### CODE REVIEW

**Comment by silverwind** [comment]
Good change, thanks.

File: .gitignore
Comment (silverwind) [inline .gitignore]: Could remove the old entries.

Comment (wxiaoguang) [reply .gitignore]: After this PR gets approved (lgtm/done)

Otherwise, when I switch branches, there are always a lot of inconsistent "ignored" files.

----

Done

**Comment by silverwind** [comment]
There are many references to `public/img` in docs which need to be updated. Unfortunately, I think this is also breaking for all customization images.

**Comment by wxiaoguang** [comment]
> There are many references to `public/img` in docs which need to be updated.


After getting approvals, I will update the documents (including: customization, public/css, public/js, public/img)

1.21 is at the early stage, so I think we have enough time to test this PR after it gets merged.

**Comment by wxiaoguang** [comment]
CI passes.

If the code change looks good to reviewers, I will update the documents (including: customization, public/css, public/js, public/img).

If there are concerns, let's resolve the concerns first before the documentation work (to save everyone's time)

**Review by delvh — APPROVED** [review]

**Review by denyskon — APPROVED** [review]

**Review by wolfogre — APPROVED** [review]

**Review by silverwind — APPROVED** [review]

**Pull request merged**
